UnityProject/Assets/Scripts/Player/PlayerMove.cs

```diff
@@ -443,10 +445,30 @@ private void SyncWalkSpeed(float oldSpeed, float newSpeed)
 
 	public void CallActionClient()
 	{
-		if (CanUnBuckleSelf())
+		if (!CanUnBuckleSelf())
 		{
-			CmdUnbuckle();
+			return;
 		}
+
+		if (IsCuffed)
+		{
+			Chat.AddActionMsgToChat(
+				playerScript.gameObject,
+				"You're trying to ubuckle yourself from the chair! (this will take some time...)",
+				playerScript.name + " is trying to ubuckle themself from the chair!"
+			);
+			StandardProgressAction.Create(
+				new StandardProgressActionConfig(StandardProgressActionType.Unbuckle),
+				Unbuckle
+			).ServerStartProgress(
+				buckledObject.RegisterTile(),
+				buckledObject.GetComponent<BuckleInteract>().ResistTime,
+				playerScript.gameObject
+			);
+			return;
+		}
```

Comment:
Its just that this stuff reads to me as being serverside code, which is being called on the client so wont work.

Reply:
You just need to move all that code into the CmdUnbuckle()

Reply:
Can I just move this to `CmdUnbuckle()` without having to build the game? (because it takes a REALLY HUGE AMOUNT OF TIME without an SSD drive). Or is it mandatory?

Reply:
Its best to test stuff like this so broken stuff doesnt get merged

Reply:
But since this is minor change, you should be fine

Reply:
Understood, thanks. Then I'll test and migrate the code in about a day or a little more(I hope still a day). So it's not safe to merge this PR yet, carefully

Reply:
Done. I moved the code and tested it all on a headless server with two clients, everything seems to work well